connectors/sources/outlook.py

```diff
@@ -0,0 +1,916 @@
+                )
+            )
+        await self.queue.put(END_SIGNAL)
+
+    async def _fetch_tasks(self, account, timezone):
+        async for task in self.client.get_tasks(account=account):
+            await self.queue.put(
+                (
+                    self.doc_formatter.task_doc_formatter(task=task, timezone=timezone),
+                    None,
+                )
+            )
+
+            if task.has_attachments:
+                await self.fetchers.put(
+                    partial(
+                        self._fetch_attachments, outlook_object=task, timezone=timezone
+                    )
+                )
+                self.tasks += 1
+
+        await self.queue.put(END_SIGNAL)
+
+    async def _fetch_calendars(self, account, timezone):
+        async for calendar in self.client.get_calendars(account=account):
+            await self._enqueue_calendars(
+                calendar=calendar, child_calendar=calendar, timezone=timezone
+            )
+
+        await self.queue.put(END_SIGNAL)
+
+    async def _fetch_child_calendars(self, account, timezone):
+        async for calendar, child_calendar in self.client.get_child_calendars(
+            account=account
+        ):
+            await self._enqueue_calendars(
+                calendar=calendar, child_calendar=child_calendar, timezone=timezone
+            )
+
+        await self.queue.put(END_SIGNAL)
+
+    async def _enqueue_calendars(self, calendar, child_calendar, timezone):
+        await self.queue.put(
+            (
+                self.doc_formatter.calendar_doc_formatter(
+                    calendar=calendar,
+                    child_calendar=str(child_calendar),
+                    timezone=timezone,
+                ),
+                None,
+            )
+        )
+
+        if calendar.has_attachments:
+            await self.fetchers.put(
+                partial(
+                    self._fetch_attachments, outlook_object=calendar, timezone=timezone
+                )
+            )
+            self.tasks += 1
+
+    async def ping(self):
+        """Verify the connection with Outlook"""
+        await self.client.ping()
+        self._logger.info("Successfully connected to Outlook")
+
+    async def _consumer(self):
+        """Async generator to process entries of the queue
+
+        Yields:
+            dictionary: Documents from Outlook.
+        """
+        while self.tasks > 0:
+            _, item = await self.queue.get()
+            if item == END_SIGNAL:
+                self.tasks -= 1
+            else:
+                yield item
+
+    async def get_docs(self, filtering=None):
+        """Executes the logic to fetch outlook objects in async manner
+
+        Args:
+            filtering (Filtering): Object of class Filtering
+
+        Yields:
+            dictionary: dictionary containing meta-data of the files.
+        """
+        async for account in self.client._get_client.get_user_accounts():
```

Comment:
Why do we need to call `get_user_accounts` via `_get_client`?

Reply:
I think the method name - `_get_client` is confusing you. Here, `_get_client` returns an instance of `ExchangeUsers` or `Office365Users` and calling `get_user_accounts` through it. Let me rename it to something like `_get_user_instance`.